assignment_A2.ipynb

```diff
@@ -1339,44 +1339,65 @@
     "        \n",
     "        # YOUR CODE HERE\n",
     "        raise NotImplementedError()\n",
+    "    \n",
+    "    def children(self,parent):\n",
+    "        children = tuple()\n",
+    "        \n",
+    "        rchild = (parent+1)*2\n",
+    "        lchild = rchild - 1\n",
+    "        size = self.size()\n",
     "        \n",
+    "        if rchild < size: #appending right child\n",
+    "            children += (rchild,)\n",
+    "        if lchild < size: #appending left child\n",
+    "            children += (lchild,) \n",
+    "        return children\n",
```

Comment:
Determining children of a given parent index (and immediately checks if they exist)